### PR TITLE
add iron-kit metrics implementation

### PR DESCRIPTION
**Pull request by damianxu88**

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+//
+// All histograms are created with a standard set of buckets which are:
+//
+//	[0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1.0, 2.5, 5.0, 10.0, 15.0, 60.0, 300.0, 1200.0, +Infiinity]
+//
+// See the [histogram docs] for more information
+//
+// [histogram docs]: https://prometheus.io/docs/tutorials/understanding_metric_types/#histogram
+func AddHistogram(key string, help string, labels []string) {
+	_, ok := counters[key]
```
Comment (schowsf) [inline metrics/metrics.go:135]: `histograms[key]` rather than `counters`?

File: metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+func GinMiddleware(c *gin.Context) {
+	path := c.Request.URL.Path
+	s := time.Now()
+	c.Next()
+	e := time.Since(s)
+	status := c.Writer.Status()
+	statusLabels := map[string]string{statusLabel: fmt.Sprintf("%d", status), routeLabel: path}
+	UpdateHistogram(requestDurationKey, e, statusLabels)
+	IncrementCounter(totalRequestsKey, statusLabels)
+	if status < 200 || status > 299 {
```
Comment (schowsf) [inline metrics/metrics.go:205]: Do we consider 400, 404 etc error counts?  Bad request, bad paths etc... are not really error on Sprinkler.

File: metrics/metrics.go (hunk 1 above)
Comment (damianxu88) [reply metrics/metrics.go:135]: @schowsf good catch! let me update

File: metrics/metrics.go (hunk 2 above)
Comment (damianxu88) [reply metrics/metrics.go:205]: @schowsf think these would fit the generic `http_errors_total` definition, we can probably create a custom metric to capture errors we really care about for Sprinkler

**Review by schowsf — APPROVED** [review]

**Pull request merged**